### PR TITLE
add script to verify generated api client

### LINKED ISSUE #4057: Add script to verify generated API client code
**User Story**
Script checks if the user made changes and there is a need to regenerate the API client.

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add the script to verify generated API client

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4057

```release-note
NONE
```

### CODE REVIEW

File: api/hack/verify-api-client.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+TMP_DIFFROOT=$(mktemp -d)
+
+cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
+
+"${SCRIPT_ROOT}/hack/gen-api-client.sh"
+
+echo "diffing ${DIFFROOT} against freshly generated api client"
+ret=0
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
```
Comment (alvaroaleman) [inline api/hack/verify-api-client.sh]: I wouldn't expect a `verify` script to overwrite anything

Comment (alvaroaleman) [reply api/hack/verify-api-client.sh]: Also you can just remove the `|| ret=$?` and setting it to `0` on line 26 and instead just do a `ret=$?` directly after the diff command

**Review by alvaroaleman — COMMENTED** [review]
Some nitpicks, I do realize this is copied from the existing script, would you mind to update the existing script to also not do changes and verify only?

works fine thought, thanks for doing this!

File: api/hack/verify-api-client.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+API_DIR="$(go env GOPATH)/src/github.com/kubermatic/kubermatic/api"
+DIFFROOT="${API_DIR}/pkg/test/e2e/api/utils/apiclient"
+
+TMP_DIFFROOT=$(mktemp -d)
+
+cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
+
+"${SCRIPT_ROOT}/hack/gen-api-client.sh"
```
Comment (alvaroaleman) [inline api/hack/verify-api-client.sh]: Somehow my other comment got lost: Currently the codegen generates an awful lot of output that makes seeing the actual error hard, would you mind to:
* Remove the `set -x`
* Suffix line 23 with a `&>/dev/null` to discard the output of the client generation?

File: api/hack/verify-api-client.sh (hunk 1 above)
Comment (zreigz) [reply api/hack/verify-api-client.sh]: when files are different the pure command `diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}"` exit with `1` and next lines are never executed.

File: api/hack/verify-api-client.sh (hunk 2 above)
Comment (zreigz) [reply api/hack/verify-api-client.sh]: done

**Comment by zreigz** [comment]
@alvaroaleman PTAL

**Pull request merged**